assign2/src/model/filter/actions.ts

```diff
@@ -0,0 +1,23 @@
+import {FilterActions, QuestionFilter, SET_FILTER, SET_SEARCHED_TAG, SET_SEARCHED_TITLE} from "./types";
+import Tag from "../objects/Tag";
+
+export function doSetFilter(filter: QuestionFilter): FilterActions {
```

Comment:
Even though it is perfectly fine like this, as a general remark: you could have narrowed even more the return type if you exposed the component interfaces of the FilterActions union (SetFilterAction, etc) and used them instead of the FilterActions.